questionnaire/views.py

```diff
@@ -0,0 +1,61 @@
+# from django.shortcuts import get_object_or_404, redirect, render
+from django.urls import reverse_lazy
+from django.views.generic.edit import CreateView
+
+from .forms import QuestionnaireForm
+from .models import Answer, Questionnaire
+
+# from fitapp.utils import is_integrated
+
+
+# class QuestionnaireView(FormView):
+#     template_name = "questionnaire/questionnaire.html"
+#     form_class = Questionnaire
+#     success_url = reverse_lazy("answer-add")
+
+#     def form_valid(self, form):
+#         # form.save()
+#         return super().form_valid(form)
+
+# def get(self, request, pk):
+#     if request.user.is_authenticated and is_integrated(request.user):
+
+#         question = get_object_or_404(Questions, pk=pk)
+
+#         # Create context dictionary for the template
+#         context = {
+#             "question": question,
+#         }
+
+#         return render(request, self.template_name, context)
+
+#     else:
+#         redirect("login")
+
+# def post(self, request):
+#     pass
+
+
+# class QuestionnaireView(CreateView):
+#     template_name = "questionnaire/questionnaire.html"
+#     success_url = reverse_lazy("questionnaire")
+#     # fields = ["question", "answer"]
+#     # context_object_name = "answers"
+
+#     def get_success_url(self, **kwargs):
+#         pass
+
+
+# ANTOINE
+class QuestionnaireView(CreateView):
```

Comment:
What I wrote was heavy pseudocode, you'll probably at least need to do something with 
`def get_success_url`, and ensure the returned url points to the next question in the questionnaire
```python
 return reverse_lazy("questionnaire". args={"question" : int(kwargs['question'])+1})
```

Reply:
I sorted that out like this:
```python
    def get_success_url(self):
        question_id = self.kwargs["question"] + 1
        questionnaire_pk = self.kwargs["questionnaire_pk"]
        max_question = Question.objects.filter(
            questionnaire_id=questionnaire_pk
        ).count()

        # Return to questionnaire with the next question
        if question_id < max_question:
            return reverse_lazy(
                "questionnaire",
                kwargs={"questionnaire_pk": questionnaire_pk, "question": question_id},
            )

        # Return to home when all the questions have been answered
        elif question_id == max_question:
            return reverse_lazy("home")
```

Not sure this is neither the cleanest nor best practices way to do it but that works.
Advice is welcome if you have a better approach in mind :)